File: server/Models/PcBuild.cs
namespace PCpals.Models;

public class PcBuild : RepoItem<int>{
    public string Name {get; set;}
    public string CreatorId {get; set;}
    public int PowerScore {get; set;}
    public decimal Price {get; set;}
    public int Rating {get; set;}
    public Boolean IsPrivate {get; set;}
    public int Watts {get; set;}
    public string PcCase {get; set;}
    public string CasePicture {get; set;}
    public string PcCpu {get; set;}
    public string Gpu {get; set;}
    public string MotherBoard {get; set;}
    public string Ram {get; set;}
    public string PcStorage {get; set;}
    public string PowerSupply {get; set;}
    public string Cooler {get; set;}
    public Account Creator {get; set;}
}

File: client/src/models/PcBuild.js

export class PcBuild{
    constructor(data){
        this.id =data.id
        this.name = data.name
        this.creatorId = data.creatorId
        this.powerScore = data.powerScore || 0
        this.price = data.price || 0
        this.rating = data.rating || 0
        this.creator = data.creator
        this.isPrivate = data.isPrivate
        this.watts = data.watts
        this.maxWattage = data.maxWattage || 0
        this.pcCase = data.pcCase
        this.casePicture = data.casePicture
        this.pcCpu = data.pcCpu
        this.gpu = data.gpu
        this.motherBoard = data.motherBoard
        this.ram = data.ram
        this.pcStorage = data.pcStorage
        this.cooler = data.cooler
        this.powerSupply = data.powerSupply
    }
}